c/parallel/src/scan.cu

```diff
@@ -0,0 +1,589 @@
+#include <cub/agent/single_pass_scan_operators.cuh>
+#include <cub/detail/choose_offset.cuh>
+#include <cub/detail/launcher/cuda_driver.cuh>
+#include <cub/device/dispatch/dispatch_scan.cuh>
+#include <cub/thread/thread_load.cuh>
+#include <cub/util_arch.cuh>
+#include <cub/util_temporary_storage.cuh>
+#include <cub/util_type.cuh>
+
+#include <format>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <type_traits>
+
+#include "cub/util_device.cuh"
+#include "kernels/iterators.h"
+#include "kernels/operators.h"
+#include "util/context.h"
+#include "util/errors.h"
+#include "util/indirect_arg.h"
+#include "util/types.h"
+#include <cccl/c/scan.h>
+#include <nvrtc.h>
+#include <nvrtc/command_list.h>
+
+struct op_wrapper;
+struct device_scan_policy;
+using OffsetT = unsigned long long;
+static_assert(std::is_same_v<cub::detail::choose_offset_t<OffsetT>, OffsetT>, "OffsetT must be size_t");
+
+struct input_iterator_state_t;
+struct output_iterator_t;
+
+namespace scan
+{
+
+struct scan_runtime_tuning_policy
+{
+  int block_size;
+  int items_per_thread;
+  cub::CacheLoadModifier load_modifier;
+
+  scan_runtime_tuning_policy Scan() const
+  {
+    return *this;
+  }
+
+  int ItemsPerThread() const
+  {
+    return items_per_thread;
+  }
+
+  int BlockThreads() const
+  {
+    return block_size;
+  }
+
+  cub::CacheLoadModifier LoadModifier() const
+  {
+    return load_modifier;
+  }
+};
+
+template <typename Tuning, int N>
+Tuning find_tuning(int cc, const Tuning (&tunings)[N])
+{
+  for (const Tuning& tuning : tunings)
+  {
+    if (cc >= tuning.cc)
+    {
+      return tuning;
+    }
+  }
+
+  return tunings[N - 1];
+}
+
+scan_runtime_tuning_policy get_policy(int /*cc*/, cccl_type_info /*accumulator_type*/)
+{
+  // TODO(ashwin): returning some default for now:
+  return {128, 15, cub::LOAD_CA};
```

Comment:
To be honest, I was kind of overwhelmed by everything going on in https://github.com/NVIDIA/cccl/blob/main/cub/cub/device/dispatch/tuning/tuning_scan.cuh. Could use some guidance on how best to reasonably implement `get_policy` here.

Reply:
For now, this is fine. We will tackle the larger problem of unifying tuning definitions between C++ and c.parallel in the future.